### PR TITLE
chore: [#176841424,IA-130] Replace backend status type with remote definition

### PR DESCRIPTION
**Pull request by Undermaken**
### ⚠️ this PR depends on https://github.com/pagopa/io-services-metadata/pull/403
This PR replaces the backend status type definition with the relative remote one

### CODE REVIEW

File: ts/features/bonus/bpd/saga/orchestration/__tests__/activateBpdOnNewPaymentMethods.test.ts
Comment (pietro909) [inline ts/features/bonus/bpd/saga/orchestration/__tests__/activateBpdOnNewPaymentMethods.test.ts:15]: Just to be clear: these are generated [io-services-metadata specs](from https://github.com/pagopa/io-services-metadata/pull/403/files#diff-f48ac1631a77bcfdff16e04799f86252e7f28d2f6ea1a7f4bc016097fd9710efR278). What is changing in this scope is not the returned value from the server but how the definitions are injected in the client.
I shouldn't require any special version of `io-dev-server` to work, correct?

File: ts/store/reducers/__mock__/backendStatus.ts
Comment (pietro909) [inline ts/store/reducers/__mock__/backendStatus.ts:155]: There are a few other `level: "warning"` in [ts/store/reducers/__tests__/backendStatus.test.ts](https://github.com/pagopa/io-app/pull/3221/files#diff-1f1ef98955c8528b761adad8533a0ac2e15e4d90285cf44ab2ae9b048b7cad0eR82) that we shall update.

File: ts/features/bonus/bpd/saga/orchestration/__tests__/activateBpdOnNewPaymentMethods.test.ts
Comment (Undermaken) [reply ts/features/bonus/bpd/saga/orchestration/__tests__/activateBpdOnNewPaymentMethods.test.ts:15]: Yes you shouldn't.
These changes are only about type definitions (before manually, now automatically generated)

Note: this changes reflect the transitions of some fields from optional to required
see https://github.com/pagopa/io-services-metadata/pull/403#discussion_r667925554

File: ts/store/reducers/__mock__/backendStatus.ts
Comment (Undermaken) [reply ts/store/reducers/__mock__/backendStatus.ts:155]: I just checked. Are you sure to have the last commit of this branches?
That file should be right sync with new spec
https://github.com/pagopa/io-app/blob/aa8987e365f08318d140630531aea62645ce96a3/ts/store/reducers/__mock__/backendStatus.ts
Otherwise I guess we should had a complire error 

Could you confirm?

Comment (pietro909) [reply ts/store/reducers/__mock__/backendStatus.ts:155]: Just checked with the latest commit: I can see many examples of `level: "warning"` in `backendStatus.test.ts`.
It works fine and runs the tests correctly, though since you're after this task you might want to replace all of them with the new enum.

Comment (Undermaken) [reply ts/store/reducers/__mock__/backendStatus.ts:155]: you are totally right.
I didn't find what you mean.
This commit should resolve (I also found a point where this field is compared incorrectly)
https://github.com/pagopa/io-app/pull/3221/commits/1dc9f3d8c2ff0349f197322fc019ba0b917091d8

**Review by fabriziofff — APPROVED** [review]

**Comment by fabriziofff** [comment]
LGFM, @pietro909 waiting for your approval for your raised points.

**Review by pietro909 — APPROVED** [review]
https://github.com/pagopa/io-services-metadata/pull/403 was merged already.
This one looks good and I'm gonna merge it.

**Pull request merged**
